src/entity/projectile/IceBomb.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\projectile;
+
+use pocketmine\block\Block;
+use pocketmine\block\VanillaBlocks;
+use pocketmine\block\Water;
+use pocketmine\event\entity\ProjectileHitEvent;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\RayTraceResult;
+use pocketmine\math\Vector3;
+use pocketmine\network\mcpe\protocol\types\entity\EntityIds;
+use pocketmine\world\particle\ItemBreakParticle;
+use pocketmine\world\sound\BlockBreakSound;
+use const PHP_INT_MAX;
+
+class IceBomb extends Throwable{
+	public static function getNetworkTypeId() : string{ return EntityIds::ICE_BOMB; }
+
+	public function getResultDamage() : int{
+		return -1;
+	}
+
+	protected function calculateInterceptWithBlock(Block $block, Vector3 $start, Vector3 $end) : ?RayTraceResult{
+		if($block instanceof Water){
```

Comment:
I think you can use getTypeId() for that

Reply:
`Water` class could be reused for other stuff which isn't supposed to behave the same way, but in general, `instanceof` should be avoided. It's not a safe way to determine a block's type.